libq/tree.h

```diff
@@ -99,6 +99,13 @@ struct tree_pkg_meta {
 	char *Q_PROPERTIES;
 	char *Q_BDEPEND;
 	/* binpkgs/vdb */
+	//
+	char *Q_CPV;
+	char *Q_PATH;
+	char *Q_BUILD_ID;
+	char *Q_BUILD_TIME;
+	char *Q_REQUIRES;
```

Comment:
from all of these, you only use Q_PATH

I know you wanted to have the complete set so others don't have to look for it, but I rather save memory and not have pointers for keys that aren't ever used

Reply:
i have made the requested changes. but I plan on using the BUILD_ID and BUILD_TIME next, they are directly relevant to picking the right multiple version.

Reply:
Do you want to make all of these available, or just the latest?  The problem you'll be facing is that the version remains the same (adding the build id would make the atom invalid).  If you only want to keep the latest, you simply have to replace the previous one, if you want to expose all versions, you'll need a hack so atom_sort will be able to order them somehow.  This will get ugly quickly I think.

Reply:
im not 100% sure what the question means, but I can say that is basically why i havent done that part yet. however, choosing any build_ID version is better than none, so i think this should be committed first.

binpkg-multi-instance has become the default in new installs since Bug #833582 was fixed for good 2 weeks ago, and the majority 90%? use case is most people will have 1 build per version. so it should be fixed soon so qmerge can work again.

after that, only the latest build_ID could be chosen by default instead, and after that, we can work on people selecting a build_ID version as part of the user input atom matching. Portage knows how to do this, so we could read that and find out how. I understand the function structure of tree.c does currently not lend well but nothings impossible.

Reply:
Ok, I'd like to go as far as just selecting the highest build_id, would that make sense?  Lacking any specification of this format (I filed a bug for that long ago) I'm assumign the build_id is a number and we can just numerically compare it to the highest version, right?